include/revng/PipelineC/PipelineC.h

```diff
@@ -149,6 +156,14 @@ void rp_manager_destroy(rp_manager *manager);
  */
 uint64_t rp_manager_containers_count(rp_manager *manager);
 
+/**
+ * Applies the diff to the model and triggers a ModelInvalidationEvent
+ *
+ * \param manager cannot be null.
+ * \param diff cannot be null.
+ */
+void rp_apply_model_diff(rp_manager *manager, const rp_model_diff *diff);
```

Comment:
This API doesn't make much sense without an API to extract the model and compute the `diff`.

Reply:
yeah, it is just missing the thing about reading diffs on which i was working, but that is way less trivial than expected

Reply:
ACK, let's move this to the invalidation PR.

Reply:
Nice we're getting to this point!
However, as previously said, this API doesn't make much sense without an API to extract the model and compute the diff.
I'd expect something like:
```c++
 void rp_model_diff_apply(const rp_model_diff *diff, rp_manager *manager);
// Or...
 void rp_manager_apply_diff(rp_manager *manager, const rp_model_diff *diff);

// Model diff I/O to make the API usable from Python
const char *rp_model_diff_serialize(const rp_model_diff *diff);
rp_model_diff *rp_model_diff_create(const char *);
```